### PR TITLE
Use config-relative path for routing annotation resources

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| License       | MIT

Requires https://github.com/symfony/symfony/pull/25113

### CODE REVIEW

File: doctrine/annotations/1.0/config/routes/annotations.yaml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 controllers:
-    resource: ../src/Controller/
+    resource: ../../src/Controller/
```
Comment (Nyholm) [inline doctrine/annotations/1.0/config/routes/annotations.yaml]: Couldn't we be using the following instead: 

```
controllers:
    resource: "%kernel.project_dir%/src/Controller"
```
???

Comment (nicolas-grekas) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: was my first idea, but parameters are not parsed in the config area, would be a whole new feature

**Comment by Nyholm** [comment]
When testing this change I get an error. 

Using `../src/Controller` works just fine. 

screen shot 2017-11-23 at 12 52 19

File: doctrine/annotations/1.0/config/routes/annotations.yaml (hunk 1 above)
Comment (Nyholm) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: Ah, okey. Thanks

**Comment by nicolas-grekas** [comment]
@Nyholm yep it works, but it's WTF (why a single "..", doesn't make any sense unless you know some internal details).
Fixed by symfony/symfony#25113 (still pending.)

File: doctrine/annotations/1.0/config/routes/annotations.yaml (hunk 1 above)
Comment (fabpot) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: As the related change is done in 3.4, it means that this config will break on project using 3.3, right?

Comment (nicolas-grekas) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: right, so I'll rebase against 3.3 as bug fix!

Comment (nicolas-grekas) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: they won't: the fallback mechanism will look on the previous path and work as before

Comment (fabpot) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: ok, great then.

Comment (fabpot) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: IIRC, this was what we had at first and it didn't work.

**Comment by nicolas-grekas** [comment]
votes pending

**Pull request merged**